### ISSUE TITLE
consider ability to add comments/attachments

### ISSUE THREAD

**Issue by allisonmuise** [Esri/crowdsource-manager#93]
use case:
responding to incidents with details and documents supporting the resolution of the report

**Comment by allisonmuise** [comment]
(Marshfield)
Adding related records to existing reports is critical for tracking and issue-response workflows. Not all info will be added from the field. Should also be able to add attachments to these related records.

**Comment by CTLocalGovTeam** [comment]
@allisonmuise ,

We are planning to provide 'Add Comment' button in comments tab as shown below
![image](https://cloud.githubusercontent.com/assets/2959186/16686556/848956de-4530-11e6-9610-6f1cbc87bbad.png)

Behaviour of the the functionality would be as described below :
1. Comments tab would always be visible, so that user would add comments even if there are no comments to edit
2. On click of 'Add Comment' button, comment form would be open with save & cancel button
3. After comment is added, message will be shown as 'Comments saved successfully' and list of comment would be displayed with this latest comment shown on top.

Please let us know your thoughts on this.

--Pratik Gohil

**Comment by allisonmuise** [comment]
@CTLocalGovTeam 

I am going to consult with the design team on this approach.

Please add the 'Add Attachment' Functionality to the Comments form, as well, if the comment table supports attachments.

**Comment by CTLocalGovTeam** [comment]
@allisonmuise

OK. We are analysing this enhancement and will keep you updated.

--Pratik Gohil

**Comment by CTLocalGovTeam** [comment]
@allisonmuise ,

Are there any update from design team on UI for adding comments/attachments ?

--Pratik Gohil

**Comment by allisonmuise** [comment]
@CTLocalGovTeam : I should get something today or Monday

**Comment by CTLocalGovTeam** [comment]
@allisonmuise,

Do we need to consider viewing of attachments in comments panel ?
If yes, could you please also take this into consideration(UI) for this while consulting with your design team.

Since, we are displaying popup media, attachment & field images into media panel do we also need to display image attachments of comments into media panel. This can quickly become a performance bottleneck if a feature has multiple comments with multiple attachments.

Please let us know your thoughts on this.

--Pratik Gohil

**Comment by allisonmuise** [comment]
@CTLocalGovTeam - OK I will think on this and let you know.

**Comment by allisonmuise** [comment]
@CTLocalGovTeam : 
For the button, let's remove the color and outline, use a grey font, and add a + in front of the text. The text should say 'Add Record. 
(see mockup sent via email)

**Comment by CTLocalGovTeam** [comment]
@allisonmuise,

OK. We are analysing this enhancement as per mockup and will keep you updated.

--Pratik Gohil

**Comment by CTLocalGovTeam** [comment]
@allisonmuise,

As suggested that 'some of the theme color' should get displayed at  '+ Add Record'  area in comment panel(sent via email)
We are planning to change the color of  '+ Add Record'  to theme color on mouse hover.


![untitled](https://cloud.githubusercontent.com/assets/2959186/17021900/9573df22-4f68-11e6-94cd-388010773363.png)


Please let us know your thoughts on this.

--Pratik Gohil

**Comment by allisonmuise** [comment]
@CTLocalGovTeam Can you share a URL to an app with this UI change? It's ok if there are other pieces that don't work yet and no testing is required - I know we are at the beginning of a sprint. I'm just looking for an actual app that I can take to the design team so we can play with some ideas in the context of the rest of the app UI.

Please don't implement the change on hover at this time, but please do change the text to read 'Add Record'

**Comment by CTLocalGovTeam** [comment]
@allisonmuise,

Application can be accessed [here](http://csslesxi9-vm3.cybertech.com/CSManager/V1.15/index.html)

Following functionalities & UI can be tested

1 UI of '+ Add Record' in comments panel

2 Viewing of comments & its attachments if any

3 Adding comments

4 Adding attachments in comments if comments table have attachment facility

5 UI of total number of comments in bracket

Note:-
'usePopupConfigurationForComment' flag is set to true in config file

--Pratik Gohil

**Comment by CTLocalGovTeam** [comment]
@allisonmuise,

We have noticed that below mockup does not contain 'Attachment' text & horizontal line below it while displaying attachments
![mockup](https://cloud.githubusercontent.com/assets/2959186/17091277/9967ceae-5256-11e6-9485-7615855a1833.png)

Should we display 'Attachments' text & horizontal line while displaying attachment in comments panel ? Sample as shown below
![96203046-50d1-11e6-892e-2fa9f3200aa1](https://cloud.githubusercontent.com/assets/2959186/17091318/f8c195f6-5256-11e6-989b-fff4728c1c06.PNG)

We do this in popup tab while displaying attachments in CrowdSource Manager app.

Please share your thoughts on this.

--Pratik Gohil

**Comment by allisonmuise** [comment]
Yes, please include the text and line. That was an oversite on my part. The UI should follow what exists on the Popup tab.

**Comment by allisonmuise** [comment]
@CTLocalGovTeam Thanks, Pratik, that will help us iterate on the design a bit here.

Could the number of related records be appended to the name of the Comments tab with only a single space between the two? 

Comments (10) 
instead of 
Comments                        (10)

**Comment by CTLocalGovTeam** [comment]
@allisonmuise,

Yes, we will append number of related records next to the name of comments tab with only a single space.

--Pratik Gohil

**Comment by allisonmuise** [comment]
@CTLocalGovTeam : Can you increase the space above and below the button to add some separation between the button and the comments? Perhaps double the space currently there?

In addition to the space, some other ideas under consideration are 
- adding a horizontal line below the button - like the one between the popup title and the popup content
- adding a very thin theme-coloured outline to the button

Could you provide mockups/screenshots/apps showing what this might look like with the extra space there, too?

Also, it is hard to judge how this will look with many comments in place as they are not currently displaying very well. I'm sure this is something you've noticed and are working on, but when this is addressed, could you update the demo app you sent?

![7-27-2016 10-35-52 am](https://cloud.githubusercontent.com/assets/2752195/17185532/5502577a-53e6-11e6-93dd-f2f615f49865.png)

**Comment by CTLocalGovTeam** [comment]
@allisonmuise,

As, suggested we have made changes to comments button & its screenshot is as shown below.

Changes made:
1 Increased space above & below the '+ Add Record' button
2 Added horizontal line below the '+ Add Record' button
3 Added a very thin theme-coloured outline to the button

Application with above changes can be accessed [here](http://csslesxi9-vm3.cybertech.com/CSManager/V1.15/index.html)

**Design 1 - Case 1** : With multiple comments, "usePopupConfigurationForComment" is set to true & comment attachments


![1](https://cloud.githubusercontent.com/assets/2959186/17211838/49983a86-54e9-11e6-908d-b4edb70d627d.PNG)

**Design 1 - Case 2** : With multiple comments, "usePopupConfigurationForComment" is set to false & comment attachments


![2](https://cloud.githubusercontent.com/assets/2959186/17211846/4f33eb66-54e9-11e6-90bf-cfe55728cf4c.PNG)


**Apart from this we also thought of some other design ideas as shown below**

**Design 2 - Case 1** : In which horizontal line below the '+ Add Record' button is removed


![3](https://cloud.githubusercontent.com/assets/2959186/17211919/ac1a6184-54e9-11e6-837f-2f0ccb9ecfea.PNG)

**Design 2 - Case 2** :


![4](https://cloud.githubusercontent.com/assets/2959186/17211920/ad294a86-54e9-11e6-9a1c-7cd3f772c4f0.PNG)

**Design 3** : On hover, border of '+ Add Record' button & '+' sign would change to theme color


![add-record-btn](https://cloud.githubusercontent.com/assets/2959186/17212100/093b896e-54eb-11e6-8683-156df5f59d93.gif)


Please share your thoughts on this & which design approach we should implement ?

--Pratik Gohil

**Comment by allisonmuise** [comment]
OK - decisions:

- [ ] Lets get rid of the horizontal line AND the button outline. 
- [ ] Center the button vertically between the top of the panel and the top of the popup title text (Leave the added space!)
- [ ] Alight the left side of the + with the left side of the popup text
- [ ] On mouseover, change the text to white and make the background of the button theme color. 
- [ ] Remove line between 'Attachments' and the thumbnails
- [ ] Make 'Attachments' font the same as the popup content font (looks smaller currently?)
- [ ] Lighten and thin line between the popup title and the popup content using #e0e0e0 and the line thickness of the current 'Attachments' line.

**Comment by CTLocalGovTeam** [comment]
@allisonmuise,

We have implemented changes suggested by you as shown below

> "Make 'Attachments' font the same as the popup content font (looks smaller currently?)"


We have analysed that both 'Attachments' & popup content font have same size & font-family. Its  looking smaller might be due to the uppercase text of popup content. We quickly tried by keeping the 'Attachments' text in uppercase & it gives same effect as popup content font.

![123](https://cloud.githubusercontent.com/assets/2959186/17247291/f220f244-55af-11e6-85e9-2dc99bcdcb60.gif)



--Pratik Gohil
